janitor/utils.py

```diff
@@ -2373,3 +2377,805 @@ def _sub_process_text_result_MultiIndex(index: pd.MultiIndex, result, df):
+                search_indices = search_indices[~rows_equal]
+
+        if search_indices.size == 0:
+            gt_left = pd.Index([])
+            gt_right = np.array([], dtype=int)
+
+        else:
+            positions = _ge_create_ranges(search_indices)
+            # get the actual index positions from right_c
+            # and map to right_keys, to get the complete view
+            right_positions = right_keys.take(positions)
+            right_indices = right_group.indices
+            gt_right = right_positions.map(right_indices)
+            gt_right = np.concatenate(gt_right)
+            gt_left = gt_left.index.repeat(search_indices)
+            # blow up left to match right_keys
+            search_indices = right_positions.map(right_group.size())
+            # blow left again to match all the index positions from right_c
+            gt_left = gt_left.repeat(search_indices)
+
+    if (
+        (lt_right.size == 0)
+        and (gt_right.size == 0)
+        and not any((right_c.hasnans, left_c.hasnans))
+    ):
+        return None, None
+
+    # nulls dont match anything, so throw it into the mix
+    right_nulls = []
+    left_nulls = []
+    if right_c.hasnans:
+        right_c_isna = right_c.isna()
+        nulls_count_right = right_c_isna.sum()
+        nulls_right = right_c.index[right_c_isna]
+        # tile is used here to ensure that every value in left_c
+        # is matched to all the nulls
+        nulls_right = np.tile(nulls_right, left_c.size)
+        right_nulls.append(nulls_right)
+        # if number of nulls from right_c is more than 1
+        # then we repeat it ... else, it is just a pairing
+        # from the right null to each of left_c
+        if nulls_count_right > 1:
+            left_nulls.append(left_c.index.repeat(nulls_count_right))
+        else:
+            left_nulls.append(left_c.index)
+
+    if left_c.hasnans:
+        left_c_isna = left_c.isna()
+        nulls_count_left = left_c_isna.sum()
+        nulls_left = left_c.index[left_c_isna]
+        # each value in right_c must be matched to all the null groups
+        nulls_left = np.tile(nulls_left, right_c.size)
+        left_nulls.append(pd.Int64Index(nulls_left))
+        # blow up right_c to match number of nulls in left
+        if nulls_count_left > 1:
+            right_nulls.append(right_c.index.repeat(nulls_count_left))
+        else:
+            right_nulls.append(right_c.index)
+
+    right_c = np.concatenate([lt_right, gt_right, *right_nulls])
+    left_c = lt_left.append([gt_left, *left_nulls])
+
+    return left_c, right_c
+
+
+def _less_than_indices(
+    left_c: pd.Series, right_c: pd.Series, strict: bool = False
+):
+    """
+    This uses binary search to get indices where left_c is less than
+    or equal to right_c. If strict is True, then only indices where `left_c`
+    is less than (but not equal to) `right_c` are returned.
+
+    Returns a tuple of (left_c, right_c)
+    """
```

Comment:
Minor grammatical nitpick.

```suggestion
    """
    Use binary search to get indices where left_c is less than
    or equal to right_c. If strict is True, then only indices where `left_c`
    is less than (but not equal to) `right_c` are returned.

    Returns a tuple of (left_c, right_c)
    """
```